gapic/src/main/com/google/gapid/widgets/LoadableImage.java

```diff
@@ -105,13 +102,10 @@ protected void onUiThreadError(Void error) {
   }
 
   public LoadableImage unload() {
-    if (--loadCount != 0) {
+    if (--loadCount != 0 || state != State.LOADING) {
       return this;
     }
-    if (state == State.LOADING) {
-      future.cancel(true);
-    }
-    updateImage(null);
```

Comment:
Why not release the image? If we ever limit the size of the cache, you'd want to do this anyway, right?